tests/cpp/test_multidevice_pipeline.cpp

```diff
@@ -217,8 +217,9 @@ DeviceMesh mesh1({1});
 DeviceMesh mesh2({0, 1, 2, 3});
 DeviceMesh mesh3({0, 2, 3});
 DeviceMesh mesh4({1, 0, 2});
-auto all_meshes = testing::Values(mesh0, mesh1, mesh2, mesh3, mesh4);
-auto all_nontrivial_meshes = testing::Values(mesh2, mesh3, mesh4);
+DeviceMesh mesh5({1, 0});
```

Comment:
Does this new test trigger the bug? I'm asking because it'd be much better to have a test to catch potential regression. Otherwise, people could accidentally remove `.contiguous` and find everything passing. If we don't have such a test, we'd better add one by creating non-contiguous tensors, e.g., `at::randn({n,m}, options).transpose()`.

Reply:
> Does this new test trigger the bug?

it does

Other tests catch it too, as you can see in the CI, but they require 3 GPUS. This one catches it with 2 GPUs only